Kernel/Devices/Audio/IntelHDA/Controller.cpp

```diff
@@ -16,13 +16,16 @@
 
 namespace Kernel::Audio::IntelHDA {
 
-UNMAP_AFTER_INIT ErrorOr<NonnullLockRefPtr<Controller>> Controller::create(PCI::DeviceIdentifier const& pci_device_identifier)
+UNMAP_AFTER_INIT ErrorOr<bool> Controller::probe(PCI::DeviceIdentifier const& device_identifier)
 {
-    auto controller_io_window = TRY(IOWindow::create_for_pci_device_bar(pci_device_identifier, PCI::HeaderType0BaseRegister::BAR0));
+    VERIFY(device_identifier.class_code().value() == to_underlying(PCI::ClassID::Multimedia));
+    return static_cast<PCI::Multimedia::SubclassID>(device_identifier.subclass_code().value()) == PCI::Multimedia::SubclassID::HDACompatibleController;
```

Comment:
Two different ways of comparing enums here: `to_underlying` vs. `static_cast`. I'd go with one or the other.